### PR TITLE
[CI] Remove user db leftovers. [Jenkinsfile] Remove useless code.

### LINKED ISSUE #876: [CI] Clean up user-find-db files
When we test DEV builds, the user-find-db files remain in the ./src/kernels directory. These should be removed before launching the next round of testing. 

[Informative] See fix for similar issue: #814.

### PR DESCRIPTION
**Pull request by atamazov**
- Resolves #876. This improves reproducibility of test results.
  - For example, #874 will be stably reproducible (or non-reproducible).
- Removes harmless merge artifacts from PR #556

### CODE REVIEW

**Comment by atamazov** [comment]
@ce1adon @shurale-nkn Please review, thanks!

**Comment by ce1adon** [comment]
Are we sure user db is the only cache that we need to clear? Can we overwrite all files in ./build?
I'm not expert here. Please add @pfultz2 as the reviewer.

**Comment by atamazov** [comment]
@ce1adon Please review 42ff550, do not worry about the rest.

File: Jenkinsfile [hunk 1]
```diff
@@ -106,6 +106,7 @@ def buildHipClangJob(Map conf, compiler){
                         mkdir build
                         rm -rf install
                         mkdir install
+                        rm -rf src/kernels/*.ufdb.txt
```
Comment (shurale-nkn) [inline Jenkinsfile:109]: pleas add src/kernels/miopen_1.0.0.udb

**Review by shurale-nkn — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (atamazov) [reply Jenkinsfile:109]: Do you really see the user-perf-db leftovers???

Comment (shurale-nkn) [reply Jenkinsfile:109]: Locally yes

Comment (atamazov) [reply Jenkinsfile]: Correct.

**Review by shurale-nkn — APPROVED** [review]

**Pull request merged**